### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by purvasharmadev**
### Changes Made:

- made cart and wishlist private routes, redirecting to login if not logged in.
- logout function.
- useEffect to store, and retrieve login status.

### files to check

auth-context.js
routes.js
cart-context.js
wishlist-context.js

### CODE REVIEW

**Review by humzahasan — APPROVED** [review]
- Easily understandable code.
- Great work with the requiredAuth component. 

Code looks good to merge!

File: src/Components/auth/login.jsx [hunk 1]
```diff
@@ -18,71 +19,65 @@ function Login() {
...
+              placeholder="********"
+            />
+            <div>
+              <label htmlFor="remember-me">
+                <input
+                  type="checkbox"
+                  name="remember-me"
+                  checked={checked}
+                  onChange={() => setChecked(!checked)}
+                  id="remember-me-check"
```
Comment (iRohitGaur) [inline src/Components/auth/login.jsx:61]: suggestion: check if `id` is still required

File: src/Context/cart-context.js [hunk 2]
```diff
@@ -69,7 +70,7 @@ function CartProvider({ children }) {
       );
       setProductCart(res.data.cart);
     } catch (error) {
-      console.log("error is: ", error);
+      console.error("error is: ", error.response.data.errors);
```
Comment (iRohitGaur) [inline src/Context/cart-context.js:73]: suggestion: toast the appropriate error

**Review by iRohitGaur — APPROVED** [review]
Nice work!

Gave some suggestions. Please look into it.

Good to merge 👍🏻

**Review by devAmlan — APPROVED** [review]
1. Code looks goood
2. Apis called and implemented successfully

Good to merge.

**Pull request merged**